### ISSUE TITLE
Add list of currencies that includes historical currencies

### ISSUE THREAD

**Issue by anton-siardziuk** [moneyphp/iso-currencies#8]
in Belarus we recently changed BYR to BYN, and imagine we have a data in database in BYR currencies and BYN currencies. And after I upgrade the library to the newest version I will no longer be able to work with old data, because Money\Money doesn't know about BYR anymore, which is wrong, as currency is there forever (as BYB - another ancient Belarus currency).

I am happy to prepare PR at some point, if maintainers say yes it makes sense to add those historical data.

**Comment by frederikbosch** [comment]
You can use a combination of ISOCurrencies, AggregateCurrencies and CurrencyList.

```php
$currencies = new AggregateCurrenices([
  new ISOCurrencies(),
  new CurrencyList(['BYR' => 2]), // the number is the number of subunits
]);
```

You can inject that where you have use the `Currencies` interface.

**Comment by anton-siardziuk** [comment]
Yeah, but this is true for everybody. When country is moved to another currency, old currency is not disappeared from historical data, which this library wouldn't be able to work with anymore.

**Comment by anton-siardziuk** [comment]
@frederikbosch could you please participate in conversation instead of just closing the issue?

**Comment by frederikbosch** [comment]
I close whatever I want. But, yes, I get your point. If you want to provide an alternative, please go ahead. A suggestion could be the following.

1. Add [three lines in fetch-update.php](https://github.com/moneyphp/iso-currencies/blob/master/bin/fetch-update.php#L13) to also save the countries as file including a date.

```php
$fetcher->saveTo(sprintf(__DIR__ . "/../resources/%s.current.php", date('Y-m-d')), new PhpExportSerializer());
$fetcher->saveTo(sprintf(__DIR__ . "/../resources/%s.current.json", date('Y-m-d')), new JsonSerializer());
$fetcher->saveTo(sprintf(__DIR__ . "/../resources/%s.current.yaml", date('Y-m-d')), new YamlSerializer());
```

2. Then create a PR @ Money package to construct ISOCurrencies from another file.
```php
ISOCurrencies::fromFile()
```

**Comment by anton-siardziuk** [comment]
My initial idea was to fetch [historical feed from here](https://www.currency-iso.org/en/home/tables/table-a3.html) as well as current feed and merge them together, so all currencies - past and present - are in the currency list and nothing is ever deleted. What do you think about this approach @frederikbosch ?

**Comment by frederikbosch** [comment]
Oh, that is not a bad idea at all.

**Comment by frederikbosch** [comment]
Then in `Money` we can still add `ISOCurrencies::fromFile` and maybe also `ISOCurrencies::historic`, both are then named constructors. The former for any file and the latter use the historic file that is generated by this feature.

**Comment by anton-siardziuk** [comment]
I have created a PR which does it in most implementation agnostic way, so decision about what to do in Money can be made later. Could you please take a look at it?

My suggestion about Money would be that by default all currencies should be used (when you do `new ISOCurrencyList()`), but if needed user should be able to use static constructors `ISOCurrencyList::current()`, `ISOCurrencyList::historic()` and `ISOCurrencyList::all()`. Default should be `all`, because I think it is what most people do - to be able to work with all the historical data no matter if this currency is still used by this country.

**Comment by frederikbosch** [comment]
Good work on the PR. Thanks! Regarding the changes in Money, it usually takes longer to get it merged. We take time to discuss new features. I would happy to help you get the suggested changes in though.

**Comment by anton-siardziuk** [comment]
Yeah, sure, I understand that changes in Money shouldn't be added so easily. That's why I tried to make those changes as flexible as possible, so we have more options with Money implementation later.

**Comment by frederikbosch** [comment]
Fixed by #9
